codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java

```diff
@@ -299,6 +299,17 @@ public static boolean isTransferEncodingChunked(HttpMessage message) {
     }
 
     /**
+     * Checks to see if {@link HttpMessage} has empty content.
+     *
+     * @param message The message to check
+     * @return True if headers indicate empty content
+     */
+    public static boolean hasEmptyContent(HttpMessage message) {
+        return !message.headers().contains(HttpHeaderNames.TRANSFER_ENCODING)
+                && getContentLength(message, 0) == 0;
```

Comment:
When `content-length` header is missing, a peer can send the content till the end of the connection. Therefore, the absence of `content-length` header should not be considered as empty content.